pygeos/constructive.py

```diff
@@ -486,6 +487,32 @@ def point_on_surface(geometry, **kwargs):
     return lib.point_on_surface(geometry, **kwargs)
 
 
+def polygonize(geometries, **kwargs):
+    """Creates polygons formed from the linework of a set of Geometries.
+
+    Polygonizes a set of Geometries which contain linework that
+    represents the edges of a planar graph. All types of Geometry are
+    accepted as input; the constituent linework is extracted as the edges
+    to be polygonized.
+
+    Parameters
+    ----------
+    geometries : array_like
```

Comment:
I'd expect this function to have an `axis` kwarg, and that to have `None` as a sensible default (see #194) - but this function is not a `ufunc.reduce` but a generalized ufunc.

Reply:
Yeah, so our other generalized ufuncs like `points` also don't have an axis keyword? 

-> That's what I wanted to answer, but actually the actual ufunc seems to support this (which I wasn't aware of, or didn't think about before):

```
In [32]: pygeos.lib.points(np.array([[1, 2], [3, 4]]), axis=1)
Out[32]: 
array([<pygeos.Geometry POINT (1 2)>, <pygeos.Geometry POINT (3 4)>],
      dtype=object)

In [33]: pygeos.lib.points(np.array([[1, 2], [3, 4]]), axis=0)
Out[33]: 
array([<pygeos.Geometry POINT (1 3)>, <pygeos.Geometry POINT (2 4)>],
      dtype=object)
```

So by default the last axis is reduced, but you can actually specify a different axis to collapse (assuming it has a correct number of elements in case of `points`)

But, those generalized ufuncs don't support the `axis=None` option.

---

Now, I could of course still add an `axis` keyword to the python-level `polygonize` function, and in the `axis=None` case pass the flattened array to the ufunc.

Reply:
I don' t think it is necessary to mimic this `axis=None` behaviour for generalized ufuncs. This is more something for numpy to implement. But forwarding `**kwargs` and documenting `axis` would be a good thing.

Reply:
>  I encountered precisely what you are suggesting in `axis=None` at the `union_all` implementation: ... maybe good to do that after all for internal API consistency

On the other hand, you could also see `polygonize` as similar to `pygeos.polygons()` (but where the input doesn't need to be strictly rings), and there the default is `axis=0`. 
Now, it can certainly be seen both ways, given that the inputs are actually "processed" in polygonize, it's maybe more alike `union_all`

Reply:
I already started with documenting the keyword. Can still add the `axis=None` functionality if we want.